Octree/include/CGAL/Octree/Node.h

```diff
@@ -0,0 +1,346 @@
+// Copyright (c) 2007-2020  INRIA (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+// Author(s)     : Jackson Campolattaro, Cédric Portaneri, Tong Zhao
+
+#ifndef CGAL_OCTREE_NODE_H
+#define CGAL_OCTREE_NODE_H
+
+#include <CGAL/license/Octree.h>
+
+#include <boost/range/iterator_range.hpp>
+
+#include <array>
+#include <memory>
+#include <bitset>
+#include <cassert>
+
+namespace CGAL {
+namespace Octree {
+namespace Node {
+
+/*!
+ * \ingroup PkgOctreeClasses
+ *
+ * \brief A class representing a single node of the tree. Alternatively referred to as a cell, octant, or subtree
+ *
+ * \details The role of the node isn't fully stable yet
+ *
+ * \tparam Point_index is the datatype the node will contain
+ */
+template<typename Point_index>
+class Node {
+
+public:
+
+  /// \name Types
+  /// @{
+
+  /*!
+   * \brief Array for containing the children of this node
+   */
+  typedef std::array<Node<Point_index>, 8> Child_list;
```

Comment:
What about children?

Reply:
I'm not sure I understand the question, do I need to make that comment more descriptive?

Reply:
I suggested to replace the word `Child_list` with `Children`.

Reply:
I agree with Andreas, especially because `list` can be misleading (you are not using a `std::list`).